src/Uno.UI/UI/Xaml/Documents/LanguageTagCache.skia.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections.Generic;
+using Windows.Globalization;
+
+#nullable enable
+
+namespace Windows.UI.Xaml.Documents;
+
+internal class LanguageTagCache
+{
+	[ThreadStatic] private static readonly Dictionary<string, string[]> _languageTagsCache = new();
+
+	public static string[] GetLanguageTag(TextElement element)
+	{
+		var key = element.Language
+			?? ApplicationLanguages.PrimaryLanguageOverride
+			?? string.Empty;
+		if (!_languageTagsCache.TryGetValue(key, out var tag))
+		{
+			var cultureInfo = key.Length == 0
+				? global::System.Globalization.CultureInfo.CurrentUICulture
+				: global::System.Globalization.CultureInfo.GetCultureInfoByIetfLanguageTag(key);
+			tag = new[] { cultureInfo.TwoLetterISOLanguageName };
+			_languageTagsCache[key] = tag;
+		}
+		return tag;
+	}
```

Comment:
I think we are caching values that may change. Is it intentional?

Also the return of this is always a single-element array. It should just be a string rather than an array.